### ISSUE TITLE
Mapping of xmlText fails

### ISSUE THREAD

**Issue by gabriel0316** [ahdis/matchbox#61]
I'm using the following CDA logical model https://gitlab.com/gabriel0316/cda-logical-model/-/blob/main/package.tgz in order to map from CDA to FHIR.

When using Matchbox 2.3.0 the following mapping script succeeds:

```
map "http://fhir.ch/ig/cda-fhir-maps/StructureMap/CdaToBundleTitle" = "CdaToBundleTitle"

uses "http://hl7.org/fhir/cda/StructureDefinition/ClinicalDocument" alias ClinicalDocument as source
uses "http://hl7.org/fhir/StructureDefinition/Bundle" alias Bundle as target

group CdaToBundleTitle(source cda : ClinicalDocument, target fhir_bundle : Bundle) {
  cda -> fhir_bundle.entry as fhir_bundle_entry_1,
    fhir_bundle_entry_1.resource = create('Composition') as fhir_composition

    then {
      cda.title as cda_title -> fhir_composition.title = (cda_title.dataString);
    }

    "CdaToBundleTitle";
}
```

The result is:

```
<?xml version="1.0" encoding="UTF-8"?>
<Bundle 
  xmlns="http://hl7.org/fhir">
  <entry>
    <resource>
      <Composition>
        <title value="Allgemeiner Laborbefund"/>
      </Composition>
    </resource>
  </entry>
</Bundle>
```

When using the current version on branch `main` `2.4.0` the mapping fails. The output is as follows:

```
<OperationOutcome 
  xmlns="http://hl7.org/fhir">
  <issue>
    <severity value="error"/>
    <code value="processing"/>
    <diagnostics value="HAPI-0389: Failed to call access method: org.hl7.fhir.exceptions.FHIRException: Exception executing transform fhir_composition.title = (cda_title.dataString) on Rule &quot;title&quot;: Index 1 out of bounds for length 1"/>
  </issue>
</OperationOutcome>
```

**Comment by oliveregger** [comment]
thanks, will be fixed with the next updated of the core libraries (see https://github.com/hapifhir/org.hl7.fhir.core/pull/1001),
you should update the FHIR Path expression to (%cda_title.dataString) (variables should be prefixed with %).

**Comment by gabriel0316** [comment]
Great! And thanks for the hint!

**Comment by oliveregger** [comment]
solved with release 3.0.0
